core/prisma/migrations/20241203164958_add_base_history_fn/migration.sql

```diff
@@ -0,0 +1,97 @@
+-- CreateEnum
+CREATE TYPE "OperationType" AS ENUM(
+    'insert',
+    'update',
+    'delete'
+);
+
+CREATE DOMAIN modified_by_type AS TEXT
+CHECK (
+    VALUE ~ '^(user|action-run|api-access-token):[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}\|\d+$|^(unknown|system)\|\d+$'
+);
```

Comment:
sad we can't specify these in prisma!
this basically checks whether the value is of the correct format. added as a `DOMAIN` type st we can always set the `lastModfiedBy` column to be this type, rather than adding a constraint for every table that uses this colum.